src/shared_utilities/SpaceVaryingInputs.jl

```diff
@@ -0,0 +1,133 @@
+# SpaceVaryingInputs.jl
+#
+# This module contains methods to process external data, regrid it onto the
+# model grid, and return the corresponding fields for use in the simulation.
+# This module only concerns with external data which varies in space,
+# and not time. For temporally varying input, we refer you to `TimeVaryingInputs.jl`.
+
+# All spatially varying parameter fields are assumed to fit into memory.
+
+# The planned parameter underlying arrays are:
+# - one-dimensional (values prescribed as a function of depth at a site),
+# - two-dimensional (values prescribed globally at each lat/lon),
+# - three-dimensional (values prescribed as a function of depth globally)
+# - analytic (functions of the coordinates of the space)
+
+module SpaceVaryingInputs
+using ClimaCore
+using DocStringExtensions
+import ..searchsortednearest
+import ..linear_interpolation
+using ClimaLand.FileReader
+export SpaceVaryingInput
+
+# Analytic case
+"""
+    SpaceVaryingInput(data_function::Function, space::S) 
+    where {S <: ClimaCore.Spaces.AbstractSpace}
+  
+Returns the parameter field to be used in the model; appropriate when
+a parameter is defined using a function of the coordinates of the space.
+
+Pass the ``data" as a function `data_function` which takes coordinates as arguments, 
+and  the ClimaCore space of the model simulation.
+"""
+function SpaceVaryingInput(
+    data_function::Function,
+    space::ClimaCore.Spaces.AbstractSpace,
```

Comment:
function and docstring don't match - maybe we want to use `where {S <: ...}` in both cases?

Reply:
I would suggest using `space::ClimaCore.Spaces.AbstractSpace` instead of `wehre {S ....`.

The difference is that the former compiles one single method for every space, the other one compiles a method for each different type. We don't need this level of specialization given that this is only run at initialization and it is not performance critical.